src/cmake/externalpackages.cmake

```diff
@@ -85,6 +85,8 @@ else ()
     checked_find_package (JPEG REQUIRED)
 endif ()
 
+checked_find_package (libuhdr REQUIRED)
```

Comment:
If it's required, we're going to need an auto-builder for it. But I would recommend that it be an optional dependency for the time being.

Reply:
I made a little build recipe that I've tested with the vfx2022 ci image:

save this as `src/cmake/build_libuhdr.cmake`:

```
# Copyright Contributors to the OpenImageIO project.
# SPDX-License-Identifier: Apache-2.0
# https://github.com/AcademySoftwareFoundation/OpenImageIO

######################################################################
# libuhdr by hand!
######################################################################

set_cache (libuhdr_BUILD_VERSION 1.2.0 "libuhdr version for local builds")
set (libuhdr_GIT_REPOSITORY "https://github.com/google/libultrahdr")
set (libuhdr_GIT_TAG "v${libuhdr_BUILD_VERSION}")

set_cache (libuhdr_BUILD_SHARED_LIBS ${LOCAL_BUILD_SHARED_LIBS_DEFAULT}
           DOC "Should execute a local libuhdr build, if necessary, build shared libraries" ADVANCED)

build_dependency_with_cmake(libuhdr
    VERSION         ${libuhdr_BUILD_VERSION}
    GIT_REPOSITORY  ${libuhdr_GIT_REPOSITORY}
    GIT_TAG         ${libuhdr_GIT_TAG}
    CMAKE_ARGS
        -D BUILD_SHARED_LIBS=${libdeflate_BUILD_SHARED_LIBS}
        -D CMAKE_INSTALL_LIBDIR=lib
    )

set (libuhdr_ROOT ${libuhdr_LOCAL_INSTALL_DIR})

find_package(libuhdr REQUIRED)

set (libuhdr_VERSION ${libuhdr_BUILD_VERSION})

if (libuhdr_BUILD_SHARED_LIBS)
    install_local_dependency_libs (uhdr uhdr)
endif ()
```

Reply:
Thanks a lot !
Should I add it even though `libuhdr` is an optional dependency ?

Reply:
ah -- you know what, if it's causing problems, maybe the build recipe should be a separate PR. 
 (although, if I'm understanding correctly, it might just be a matter of adding one more line to CMAKE_ARGS: `-D CMAKE_POSITION_INDEPENDENT_CODE=ON`

Reply:
Seems like this is fixing *almost* everything, however the ICC and Windows builds are still failing.
Should I remove these commits from the PR and create a separate one later for the build recipe ?

Reply:
Hmmmm. Might be easier to troubleshoot the build recipe if everything else were comfortably in place; and I don't want the build recipe stuff to halt progress on the rest of your PR, as it's kind of a separate problem. Up to you guys.

As for the errors, I'm realizing that I accidentally left in `-D BUILD_SHARED_LIBS=${libdeflate_BUILD_SHARED_LIBS}` instead of `${libuhdr_BUILD_SHARED_LIBS}`...! I think fixing that will permit Windows to build, install, and link dynamic libuhdr instead of static, which seems to be less troublesome for Windows.

For the ICC check, not quite sure what the fix is...

Reply:
Fine by me !
Unfortunately that didn't fix the Windows build, it looks like CMake has troubles locating the libs...
Do you know if there's a way to inspect the CI images easily ? Could be a good starting point for troubleshooting

Reply:
I don't have a very sophisticated way of testing locally - I'm working on top of a `aswf/ci-osl:2022-clang13` docker image, and looking at `src/build_scripts/gh-installdeps.bash` and `ci-startup.bash`. But I'm not sure if there's a way to test `icc` outside the CI environment...

Windows is another problem, I don't have a way to test locally. 

It's not pretty, but it might be a matter of just trying stuff and rifling through github's build logs.

The [linux-vfx2022-icc failures](https://github.com/AcademySoftwareFoundation/OpenImageIO/actions/runs/11281030497/job/31375343120?pr=4484#step:8:958) all seem to pertain to C++17 aligned allocation functions; and the problems are only occurring with icc-2022.1, not with icx-2023.1, which is interesting.

In a worst case scenario, would it be cheating if we disabled uhdr support just for the linux-vfx2022-icc check? 

Looking at https://github.com/google/libultrahdr/blob/main/docs/building.md, we could try setting `-D UHDR_ENABLE_LOGS=ON` to increase build verbosity... beyond that, I'm not sure what to suggest.

Also, it looks like there's a constraint on maximum resolution that defaults to 8192x8192. Maybe we should set `-D UHDR_MAX_DIMENSION=65535` or something?

Reply:
Okay so little investigation updates:
- Windows build failure -> looks like `libultrahdr` does not have any `install` target for Windows: https://github.com/google/libultrahdr/blob/b35a75dfbaf67e066b69f79860dfe7b429ad1485/CMakeLists.txt#L739 (or did I understand it the wrong way ?)
- ICC -> maybe the same bug with precompiled headers as in this issue: https://stackoverflow.com/questions/74541813/why-do-i-get-error-reference-to-align-val-t-is-ambiguous-with-intels-c-2
- Ubuntu gcc14 -> this one is new but from the log this doesn't seem related to the PR, no idea what's going on...

Reply:
Okay so it seems I fixed the windows build, but now the test suite is failing and the error is a bit more obscure, not sure I understand the problem

Reply:
Windows CI is fixed !
However the ICC build and now the API check are failing. @lgritz @zachlewis any ideas ?